src/main/xsl/preprocess/maprefImpl.xsl

```diff
@@ -389,6 +394,25 @@ See the accompanying LICENSE file for applicable license.
     </xsl:call-template>
   </xsl:template>
 
+  <xsl:template match="@*" mode="preserve-submap-attributes">
+    <xsl:attribute name="dita-ot:submap-{local-name()}" select="."/>
+  </xsl:template>
+  
+  <xsl:template match="*[contains(@class,' topic/title ')]" mode="preserve-submap-title-and-topicmeta">
+    <submap-topicmeta class="+ map/topicmeta ditaot-d/submap-topicmeta ">
+      <submap-title class="+ topic/navtitle ditaot-d/submap-title ">
+        <xsl:apply-templates select="@*" mode="preserve-submap-attributes"/>
+        <xsl:apply-templates select="*|processing-instruction()|text()"/>
+      </submap-title>
+    </submap-topicmeta>
+  </xsl:template>
+  <xsl:template match="*[contains(@class,' map/topicmeta ')]" mode="preserve-submap-title-and-topicmeta">
```

Comment:
I'm not sure, it raises another issue not covered by this yet.

Currently we don't save `<topicmeta>` from either the referencing map or the referenced map. 

My primary goal here was to save the title from the referenced map (it's the only thing I actually need for my current use case). I added in `topicmeta` because it's there and easy to save at the same time, using the hidden `foreign` element.

If we're keeping the original metadata from the referenced map as a literal `<topicmeta>`, it sort of highlights that we're dropping the metadata from the referencing map. If we save both, then either we have to use `foreign` for one, or we have to re-implement the metadata merge rules (I don't think we want to do that).

There seem to be a few ways we could proceed on this. In all cases I'm assuming we save the title somehow, and then:

1. Forget about topicmeta, just keep title because that's straightforward, can be put into a generated `topicmeta/navtitle`
1. Put the `topicmeta` contents from the _referenced_ map in as-is (latest suggestion above)
1. Put either the _referenced_ or the _referencing_ `topicmeta` in as-is, but store the other `topicmeta` as foreign
1. Put both the _referenced_ and _referencing_ `topicmeta` in as separate foreign elements, again just making them available for use later as needed
1. Merge the topicmeta using the normal referenced / referencing priority rules, which means re-implementing or re-using that algorithm in the `mapref` module, which I'm not excited about